test/arrayops.jl

```diff
@@ -271,18 +271,30 @@ end
     @test typeof(Vector(3)) == Vector{Any}
     @test typeof(Vector()) == Vector{Any}
     @test typeof(Matrix{Int}(2,3)) == Matrix{Int}
-    @test typeof(Matrix{Int}()) == Matrix{Int}
     @test typeof(Matrix(2,3)) == Matrix{Any}
-    @test typeof(Matrix()) == Matrix{Any}
 
     @test size(Vector{Int}(3)) == (3,)
     @test size(Vector{Int}()) == (0,)
     @test size(Vector(3)) == (3,)
     @test size(Vector()) == (0,)
     @test size(Matrix{Int}(2,3)) == (2,3)
-    @test size(Matrix{Int}()) == (0,0)
     @test size(Matrix(2,3)) == (2,3)
-    @test size(Matrix()) == (0,0)
+
+    # TODO: will throw MethodError after 0.6 deprecations are deleted
+    dw = Base.JLOptions().depwarn
+    if dw == 2
+        @test_throws ErrorException Matrix{Int}()
+        @test_throws ErrorException Matrix()
+    elseif dw == 1
+        @test_warn "deprecated" Matrix{Int}()
+        @test_warn "deprecated" Matrix()
+    elseif dw == 0
+        @test size(Matrix{Int}()) == (0,0)
+        @test size(Matrix()) == (0,0)
+    else
+        error("unexpected depwarn value")
+    end
+    @test_throws MethodError Array{Int,3}()
```

Comment:
though we probably want to put such tests in a single place and make sure they get run regularly with all 3 values of the flag (or expand to 3 tests that modify the option in a scoped way, if that's possible inside julia)